### PR TITLE
[CBR-377] Separate Plugins into 'Plugins' and 'LegacyPlugins'

### PR DESCRIPTION
**Pull request by KtorZ**
## Description

Not quite done, but this is the foundation. Still have to re-implement the update notifier plugin. I'll have a look at that ASAP, but in the meantime, this can go in if you feel like it.

## Linked issue

[[CBR-377]](https://iohk.myjetbrains.com/youtrack/issue/CBR-377)

## Type of change

- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [x] 🛠 New feature (non-breaking change which adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)
- [x] 🏭 Refactoring that does not change existing functionality but does improve things like code readability, structure etc
- [ ] 🔨 New or improved tests for existing code
- [ ] ⛑ git-flow chore (backport, hotfix, etc)

## Developer checklist

- [x] I have read the [style guide](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/style-guide.md) document, and my code follows the code style of this project.
- [x] If my code deals with exceptions, it follows the [guidelines](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/exceptions.md).
- [x] I have updated any documentation accordingly, if needed. Documentation changes can be reflected in opening a PR on [cardanodocs.com](https://github.com/input-output-hk/cardanodocs.com), amending the inline [Haddock](https://www.haskell.org/haddock/) comments, any relevant README file or one of the document listed in the [docs](https://github.com/input-output-hk/cardano-sl/tree/develop/docs) directory.
- [ ] CHANGELOG entry has been added and is linked to the correct PR on GitHub.

## Testing checklist

- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: wallet-new/src/Cardano/Wallet/Server/LegacyPlugins.hs [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+  | isDebugMode wrm = corsMiddleware app
+  | otherwise = app
+
+corsMiddleware :: Middleware
+corsMiddleware = cors (const $ Just policy)
+    where
+      policy = simpleCorsResourcePolicy
+        { corsRequestHeaders = ["Content-Type"]
+        , corsMethods = "PUT" : simpleMethods
+        }
```
Comment (KtorZ) [inline wallet-new/src/Cardano/Wallet/Server/LegacyPlugins.hs]: Same stuff as before in `Plugins` minus the `walletBackend` function.

File: wallet-new/server/Main.hs [hunk 2]
```diff
@@ -152,11 +153,24 @@ actionWithNewWallet pm txpConfig sscParams nodeParams ntpConfig params =
         -> (Diffusion Kernel.Mode.WalletMode -> Kernel.Mode.WalletMode ())
     runNodeWithInit w nr = runNode pm txpConfig nr (plugins w)
 
-    -- TODO: Don't know if we need any of the other plugins that are used
-    -- in the legacy wallet (see 'actionWithWallet').
+    -- FIXME: Do we need the monitoring API ?
```
Comment (adinapoli-iohk) [inline wallet-new/server/Main.hs]: I think we might, but I can pick this up while you on holiday, @KtorZ 😉

File: wallet-new/src/Cardano/Wallet/Server/Plugins.hs [hunk 3]
```diff
@@ -2,267 +2,134 @@
...
-            walletAddress
-            -- Disable TLS if in debug modeit .
+            (BS8.unpack ip)
+            port
             (if isDebugMode walletRunMode then Nothing else walletTLSParams)
             Nothing
-            (Just portCallback)
+            (Just $ portCallback ctx)
   where
+    (ip, port) = walletDocAddress
```
Comment (adinapoli-iohk) [inline wallet-new/src/Cardano/Wallet/Server/Plugins.hs]: @KtorZ hehe, just spend a good hour debugging an obscure error which turned out to be this seemingly-innocent binding 😁 

this ought to be `walletAddress` only.

**Comment by adinapoli-iohk** [comment]
Superseded by https://github.com/input-output-hk/cardano-sl/pull/3514.

File: wallet-new/server/Main.hs (hunk 2 above)
Comment (KtorZ) [reply wallet-new/server/Main.hs]: Apparently, ops don't use it. Maybe exchanges do?

File: wallet-new/src/Cardano/Wallet/Server/Plugins.hs (hunk 3 above)
Comment (KtorZ) [reply wallet-new/src/Cardano/Wallet/Server/Plugins.hs]: Oopsie :no_mouth:

**Pull request closed without merging**